### PR TITLE
Implement minimum time on page check

### PR DESCRIPTION
**Pull request by da2x**
* Only run on topmost document (not an iframe)
* Implement minimum time on page as synced setting (when available)

### CODE REVIEW

File: scripts/timeline.js [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 // Only run in main frame (primary document)
 if (window.parent.location == window.location) {
-    // Run when document has loaded
-    (document.readyState == 'complete' || document.readyState == 'interactive') 
-        ? CreateOrUpdateActivity() : document.addEventListener('DOMContentLoaded', CreateOrUpdateActivity);
+
+    // Get minimum time on page time
+    (chrome.storage.sync || chrome.storage.local).get('min_sec_loaded', function(data) {
+        if (data.min_sec_loaded) {
+            minimumTimeOnPage = data.access_token;
```
Comment (DominicMaas) [inline scripts/timeline.js]: Setting the minimum time to the access token?

**Comment by da2x** [comment]
May be more like 10 seconds on a slow system because of the delay when reading from chrome.storage. I couldn’t find a good way to handle the delay in the background script without involving and complicating matter greatly with the chrome.tabs API.

File: scripts/timeline.js (hunk 1 above)
Comment (da2x) [reply scripts/timeline.js]: Woopsie.

**Comment by da2x** [comment]
8 second default comes from Brave. They use it as the minimum attention-time required for a page. Probably just as good of a default as anything.

**Comment by DominicMaas** [comment]
Thanks, all looks good.

**Pull request merged**